sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -173,23 +173,28 @@ class GpuSingleDirectoryDataWriter(
       }
       currentWriter.writeAndClose(batch, statsTrackers)
     } else {
-      withResource(batch) { batch =>
+      withResource(batch) { _ =>
+        assertNotExceedMaxFileCounter(
+          fileCounter,
+          maxRecordsPerFile,
+          recordsInFile,
+          batch.numRows(),
+          MAX_FILE_COUNTER
+        )
+        val dataTypes = GpuColumnVector.extractTypes(batch)
+        var needNewWriter = recordsInFile >= maxRecordsPerFile
         withResource(GpuColumnVector.from(batch)) {table => 
-          val splitIndexes = getSplitIndexes(
-            maxRecordsPerFile,
-            recordsInFile,
-            table.getRowCount()
-          )
-  
-          val dataTypes = GpuColumnVector.extractTypes(batch)
-  
-          var needNewWriter = recordsInFile >= maxRecordsPerFile
-          withResource(table.contiguousSplit(splitIndexes: _*)) {tabs =>
+          withResource{
+            val splitIndexes = getSplitIndexes(
```

Comment:
Why to put the calling to `getSplitIndexes` inside a `withResource` ? I don't think it is better than the old way.

Reply:
We don't put `getSplitIndexes` under `WithResource` here. I just put it in the expression of `table.contiguousSplit(splitIndexes: _*)`  to shrink its lifetime